extensions/applicationinsights-debugplugin-js/src/DebugPlugin.ts

```diff
@@ -0,0 +1,295 @@
+
+    /**
+     * the style that will be permanently embedded in the webpage
+     * TODO: manage style conflicts (prepend unique ID to relevant class names?)
+     */
+    let permStyleEl: HTMLStyleElement;
+
+    /**
+     * an object containing the individual debug bin items
+     */
+    let debugBins: {
+      [key: string]: DebugBin;
+    };
+
+    /**
+     * the parent containing all the individual debugBins
+     */
+    let debugBinParent: DebugBinParent;
+
+    /**
+     * the different telemetry functions that will be tracked
+     */
+    let trackers: string[];
+
+    /**
+     * appinsights analytics extension instance. useful methods on __proto__
+     */
+    let analyticsExt: IPlugin;
+
+    /**
+     * ajax dependency extension instance. useful methods on __proto__
+     */
+    let ajaxDependencyExt: IPlugin;
+
+    /**
+     * appinsights properties extension instance. useful methods on __proto__
+     */
+    let propertiesExt: IPlugin;
+
+    /**
+     * appinsights channel extension instance.
+     */
+    let channelPluginExt: IPlugin;
+
+    /**
+     * timestamp used to track number of seconds since webpage was loaded
+     */
+    let startTime: number = +new Date();
+
+    /**
+     * the config for this plugin
+     */
+    let _extensionConfig: IDebugPluginConfig = DebugPlugin.getDefaultConfig();
+
+    dynamicProto(DebugPlugin, this, (_self, base) => {
+      _self.initialize = (config: IDebugPluginConfig, core: IAppInsightsCore, extensions: IPlugin[], pluginChain?: ITelemetryPluginChain) => {
+        if (!_self.isInitialized()) {
+          base.initialize(config, core, extensions, pluginChain);
+
+          // 1. grabs each extension involved in instrumentation
+          for (let i = 0; i < extensions.length; i++) {
+            if (extensions[i].identifier === 'ApplicationInsightsAnalytics') {
+              analyticsExt = extensions[i];
+            }
+            else if (extensions[i].identifier === 'AjaxDependencyPlugin') {
+              ajaxDependencyExt = extensions[i];
+            }
+            else if (extensions[i].identifier === 'AppInsightsPropertiesPlugin') {
+              propertiesExt = extensions[i];
+            }
+            else if (extensions[i].identifier === 'AppInsightsChannelPlugin') {
+              channelPluginExt = extensions[i];
+            }
+          }
+
+          // 2. initialize trackers if they don't exist
+          const defaultConfig = DebugPlugin.getDefaultConfig();
+          const ctx = _self._getTelCtx();
+          const identifier = _self.identifier;
+          for (const field in defaultConfig) {
+              _extensionConfig[field] = () => ctx.getConfig(identifier, field, defaultConfig[field]());
+          }
+
+          trackers = _extensionConfig.trackers();
+
+          // 3. setup debugBin
+          const debugBinContainer = document.createElement("div");
+          debugBinContainer.className = 'debug-bin-container';
```

Comment:
There is a possible name clash situation here so this is a bit generic.
Lets add "ai-" as the prefix to reduce the possibility of someone else creating their own "debug*" classes.

Reply:
Alrighty - for the sake of covering all bases I implemented that as a configuration option to give every classname a prefix (that defaults to "ai")